### PR TITLE
Enhancement/2976 analyitcs 4 asset

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #2976

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/googlesitekit-modules-analytics-4.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { registerStore } from './modules/analytics';
```
Comment (aaemnnosttv) [inline assets/js/googlesitekit-modules-analytics-4.js:23]: This references the wrong module 😄 
```suggestion
import { registerStore } from './modules/analytics-4';
```

File: assets/js/modules/index.test.js [hunk 2]
```diff
@@ -22,25 +22,37 @@
...
+	return fs.readdirSync( dir, { withFileTypes: true } )
+		.filter( ( file ) => file.isDirectory() )
+		.map( ( file ) => file.name );
+}
+
+function getComponentNames( componentPath ) {
+	return fs.readdirSync( componentPath )
+		.filter( ( name ) => ! /^index|\.test\.js$/.test( name ) )
+		.map( ( name ) => name.replace( /\..*/, '' ) );
+}
```
Comment (aaemnnosttv) [inline assets/js/modules/index.test.js:40]: Can you elaborate on why the changes here were necessary?

**Review by aaemnnosttv — COMMENTED** [review]
Looks good, just one small detail seems to be missed and a question about some of the other changes.

File: assets/js/googlesitekit-modules-analytics-4.js (hunk 1 above)
Comment (eugene-manuilov) [reply assets/js/googlesitekit-modules-analytics-4.js]: :man_facepalming: fixed!

File: assets/js/modules/index.test.js (hunk 2 above)
Comment (eugene-manuilov) [reply assets/js/modules/index.test.js:40]: The original test expects that all modules have the `components` folder and tries to check files and exports there. The new `analytics-4` module doesn't have components yet, so I have had to update this test to avoid false-positive fails.

**Review by aaemnnosttv — APPROVED** [review]
